### PR TITLE
Lint tests Shadow - Z

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
More linting of tests.

### CODE REVIEW

**Review by timmysilv — APPROVED** [review]

File: tests/tape/test_tape.py [hunk 1]
```diff
@@ -998,13 +998,14 @@ def test_depth_expansion(self):
...
                 # will be decomposed into a RZ, RY, RZ
                 qml.Rot(0.543, 0.1, 0.4, wires=0)
 
             qml.CNOT(wires=[0, "a"])
             qml.RY(0.2, wires="a")
-            qml.probs(wires=0), qml.probs(wires="a")
+            qml.probs(wires=0)
+            qml.probs(wires="a")
 
         new_tape = tape.expand(depth=3)
```
Comment (dwierichs) [inline tests/tape/test_tape.py:1010]:
```suggestion
        new_tape = tape.expand(depth=2)
```

File: tests/tape/test_tape.py [hunk 2]
```diff
@@ -35,26 +35,26 @@
...
     assert tape.observables == [obs]
 
     # now create an old QNode
     dev = qml.device("default.qubit", wires=[0, "a"])
 
     @qml.qnode(dev)
     def func(x):
-        global op
-        op = qml.RX(x, wires=0)
+        nonlocal op_
```
[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/tape/test_tape.py:50]: Above `op_` is created within a context manager, which does not have an own scope. Here it is within a function, so there is a function scope and we need to make `op_` nonlocal in order to access it outside of `func`/the corresponding `QNode`.

File: tests/tape/test_tape.py [hunk 3]
```diff
@@ -472,15 +472,15 @@ def test_iteration_preserves_circuit(self):
...
+                qml.apply(op_)
 
         # Check that the underlying circuit is as expected
         assert tape.circuit == circuit
 
         # Iterate over the tape
-        for op in tape:
-            op
+        for op_ in tape:
+            op_  # pylint: disable=pointless-statement
```
[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/tape/test_tape.py:483]: I guess we could also just `pass`? :thinking:

File: tests/tape/test_tape.py (hunk 1 above)
Comment (timmysilv) [reply tests/tape/test_tape.py]: I was more thinking to change the docstring, sorry for the added hassle

File: tests/tape/test_tape.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (timmysilv) [reply tests/tape/test_tape.py:483]: `test_tape.py` does not mean much to me since we avoid `QuantumTape` in general, so I would say follow your heart

**Review by mudit2812 — APPROVED** [review]

File: tests/tape/test_tape.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/tape/test_tape.py:483]: Doing 
```python
_ = list(tape)
```
also works, as the list instantiation would consume an iterator as well.

Resolving...

**Pull request merged**